sigmastate/src/main/scala/sigmastate/crypto/CryptoFacade.scala

```diff
@@ -0,0 +1,86 @@
+package sigmastate.crypto
+
+import java.math.BigInteger
+
+/** A facade for cryptographic operations. The concrete implementation is delegated to the
+  * Platform object, which is resolved by the compiler to either JVM or JS implementation.
+  * Cross-platform code should use this facade instead of the Platform object directly.
+  */
+object CryptoFacade {
+  /** Create a new context for cryptographic operations. */
+  def createCryptoContext(): CryptoContext = Platform.createContext()
+
+  /** * Normalization ensures that any projective coordinate is 1, and therefore that the x, y
+    * coordinates reflect those of the equivalent point in an affine coordinate system.
+    *
+    * @return a new ECPoint instance representing the same point, but with normalized coordinates
+    */
+  def normalizePoint(p: Ecp): Ecp = Platform.normalizePoint(p)
+
+  /** Negate a point. */
+  def negatePoint(p: Ecp): Ecp = Platform.negatePoint(p)
+
+  /** Check if a point is infinity. */
+  def isInfinityPoint(p: Ecp): Boolean = Platform.isInfinityPoint(p)
+
+  /** Exponentiate a point.
+    *
+    * @param p point to exponentiate
+    * @param n exponent
+    * @return p to the power of n (`p^n`)
+    */
+  def exponentiatePoint(p: Ecp, n: BigInteger): Ecp = Platform.exponentiatePoint(p, n)
+
+  /** Multiply two points.
+    *
+    * @param p1 first point
+    * @param p2 second point
+    * @return group multiplication (p1 * p2)
+    */
+  def multiplyPoints(p1: Ecp, p2: Ecp): Ecp = Platform.multiplyPoints(p1, p2)
+
+  /** Return simplified string representation of the point (used only for debugging) */
+  def showPoint(p: Ecp): String = Platform.showPoint(p)
+
+  /** Returns the value of bit 0 in BigInteger representation of this point. */
+  def testBitZeroOfFieldElem(p: ECFieldElem): Boolean = Platform.testBitZeroOfFieldElem(p)
+
+  /** Returns byte representation of the given field element. */
+  def getEncodedOfFieldElem(p: ECFieldElem): Array[Byte] = Platform.getEncodedOfFieldElem(p)
```

Comment:
encodeFieldElem ? getEncodedOfFieldElem sounds like result should be of Encoded[] type

Reply:
The logic behind this name is that it uses `getEncoded` method of ECFieldElement. So, if you are not insisting of the rename let's leave it like this.

Reply:
Why are you exposing in the name what is called inside ? The name should say what the method is doing, not how

Reply:
renamed